pkg/candlepin_client/pool.go

```diff
@@ -0,0 +1,40 @@
+package candlepin_client
+
+import (
+	"time"
+
+	caliri "github.com/content-services/caliri/release/v4"
+)
+
+func (c *cpClientImpl) FetchPool(ownerKey string) (*caliri.PoolDTO, error) {
+	found, httpResp, err := c.client.OwnerAPI.ListOwnerPools(c.ctx, ownerKey).Execute()
+	if httpResp != nil {
+		defer httpResp.Body.Close()
+	}
+	if len(found) == 0 {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, errorWithResponseBody("couldn't fetch product", httpResp, err)
+	}
+	return &found[0], nil
```

Comment:
I wonder if we should log an error if there is more than 1?

Reply:
also, are you sure that this doesn't return red hat pools?  (i'm not actually sure myself)

Reply:
locally there are no red hat pools, but I guess there normally would be? If that's the case, we'd have to store the pool ID on something in the database I think. Because we're not able to specify the pool ID during creation (we can but it doesn't work).

Since it's one pool per org, I'm thinking to store on the domains table?

Reply:
I changed this to filter by the product ID, which I think would be good enough?

Reply:
yep, i think so too!